projects/scripts/adi_project_intel.tcl

```diff
@@ -103,33 +116,35 @@ proc adi_project {project_name {parameter_list {}}} {
   puts $QFILE "set_project_property DEVICE $device"
   puts $QFILE "foreach {param value} {$parameter_list} { set ad_project_params(\$param) \$value }"
   puts $QFILE "source system_qsys.tcl"
-  puts $QFILE "set_interconnect_requirement {\$system} {qsys_mm.clockCrossingAdapter} {AUTO}"
-  puts $QFILE "set_interconnect_requirement {\$system} {qsys_mm.burstAdapterImplementation} {PER_BURST_TYPE_CONVERTER}"
-  puts $QFILE "set_interconnect_requirement {\$system} {qsys_mm.maxAdditionalLatency} {4}"
+  puts $QFILE "set_domain_assignment {\$system} {qsys_mm.maxAdditionalLatency} {4}"
+  puts $QFILE "set_domain_assignment {\$system} {qsys_mm.clockCrossingAdapter} {AUTO}"
+  puts $QFILE "set_domain_assignment {\$system} {qsys_mm.burstAdapterImplementation} {PER_BURST_TYPE_CONVERTER}"
   puts $QFILE "save_system {system_bd.qsys}"
   close $QFILE
 
-  exec -ignorestderr $quartus(quartus_rootpath)/sopc_builder/bin/qsys-script \
-    --script=system_qsys_script.tcl
-  exec -ignorestderr $quartus(quartus_rootpath)/sopc_builder/bin/qsys-generate \
-    system_bd.qsys --synthesis=VERILOG --output-directory=system_bd \
-    --family=$family --part=$device
+  # check which type of Quartus is used, to call the qsys utilities with the
+  # correct attributes
+  if {$quartus_pro_isused == 1} {
+
+    exec -ignorestderr $quartus(quartus_rootpath)/sopc_builder/bin/qsys-script \
+      --quartus_project=$project_name --script=system_qsys_script.tcl
+
+    exec -ignorestderr $quartus(quartus_rootpath)/sopc_builder/bin/qsys-generate \
+      system_bd.qsys --synthesis=VERILOG --family=$family --part=$device \
+      --quartus-project=$project_name
+
+  } else {
+
+    exec -ignorestderr $quartus(quartus_rootpath)/sopc_builder/bin/qsys-script \
+      --script=system_qsys_script.tcl
 
-  # ignored warnings and such
+    exec -ignorestderr $quartus(quartus_rootpath)/sopc_builder/bin/qsys-generate \
+      system_bd.qsys --synthesis=VERILOG --family=$family --part=$device \
 
-  set_global_assignment -name MESSAGE_DISABLE 17951 ; ## unused RX channels
-  set_global_assignment -name MESSAGE_DISABLE 18655 ; ## unused TX channels
```

Comment:
I would rather keep these critical warnings disabled or find another solution. Having critical warnings in the master branch is not desirable, even if they don't affect the design.